openverse_catalog/dags/storage/test_image.py

```diff
@@ -0,0 +1,290 @@
+        return source
+
+    monkeypatch.setattr(util, "get_source", mock_get_source)
+
+    def mock_enrich_tags(tags):
+        return tags
+
+    monkeypatch.setattr(image_store, "_enrich_tags", mock_enrich_tags)
+
+    actual_image = image_store._get_image(**args_dict)
+    args_dict["tags"] = args_dict.pop("raw_tags")
+    args_dict["provider"] = "testing_provider"
+    args_dict["filesize"] = None
+    args_dict["license_"] = args_dict.get("license_info").license
+    args_dict["license_version"] = args_dict.pop("license_info").version
+
+    assert actual_image == image.Image(**args_dict)
+
+
+@pytest.fixture
+def default_image_args(
+    setup_env,
+):
+    return dict(
+        foreign_identifier=None,
+        foreign_landing_url="https://image.org",
+        image_url="https://image.org",
+        thumbnail_url=None,
+        width=None,
+        height=None,
+        filesize=None,
+        license_="cc0",
+        license_version="1.0",
+        creator=None,
+        creator_url=None,
+        title=None,
+        meta_data=None,
+        tags=None,
+        watermarked=None,
+        provider=None,
+        source=None,
+        ingestion_type="provider_api",
+    )
+
+
+def test_create_tsv_row_non_none_if_req_fields(
+    default_image_args,
+    setup_env,
+):
+    image_store = image.ImageStore()
+    test_image = image.Image(**default_image_args)
+    actual_row = image_store._create_tsv_row(test_image)
+    assert actual_row is not None
+
+
+def test_create_tsv_row_none_if_no_foreign_landing_url(
+    default_image_args,
+    setup_env,
+):
+    image_store = image.ImageStore()
+    image_args = default_image_args
+    image_args["foreign_landing_url"] = None
+    test_image = image.Image(**image_args)
+    expect_row = None
+    actual_row = image_store._create_tsv_row(test_image)
+    assert expect_row == actual_row
+
+
+def test_create_tsv_row_none_if_no_license(
+    default_image_args,
+    setup_env,
+):
+    image_store = image.ImageStore()
+    image_args = default_image_args
+    image_args["license_"] = None
+    test_image = image.Image(**image_args)
+    expect_row = None
+    actual_row = image_store._create_tsv_row(test_image)
+    assert expect_row == actual_row
+
+
+def test_create_tsv_row_none_if_no_license_version(
+    default_image_args,
+    setup_env,
+):
+    image_store = image.ImageStore()
+    image_args = default_image_args
+    image_args["license_version"] = None
+    test_image = image.Image(**image_args)
+    expect_row = None
+    actual_row = image_store._create_tsv_row(test_image)
+    assert expect_row == actual_row
+
+
+def test_create_tsv_row_returns_none_if_missing_image_url(
+    default_image_args,
+    setup_env,
+):
+    image_store = image.ImageStore()
+    image_args = default_image_args
+    image_args["image_url"] = None
+    test_image = image.Image(**image_args)
+    expect_row = None
+    actual_row = image_store._create_tsv_row(test_image)
+    assert expect_row == actual_row
```

Comment:
Since these tests are identical save for which field is being tested, this is the perfect candidate for making a [parametrized test](https://docs.pytest.org/en/6.2.x/parametrize.html)!
```suggestion
@pytest.mark.parametrize("missing_field", [
    "foreign_landing_url",
    "license_",
    "license_version",
    "image_url",
])
def test_create_tsv_row_none_if_missing_required(
    default_image_args,
    setup_env,
    missing_field,
):
    image_store = image.ImageStore()
    image_args = default_image_args
    image_args[missing_field] = None
    test_image = image.Image(**image_args)
    expect_row = None
    actual_row = image_store._create_tsv_row(test_image)
    assert expect_row == actual_row
```

That way we only have to write one test, but all of our conditions get tested :smile: What do you think?

Reply:
Thank you for adding the suggestion! I agree this is much cleaner. We have some parametrized tests, eg. in `openverse_catalog/dags/util/loader/test_sql.py`, and should try to use them in cases like this.